### PR TITLE
[#265] upgrade aws-amplify and implement static login

### LINKED ISSUE #265: Upgrade `aws-amplify` to v6

### PR DESCRIPTION
**Pull request by radoslavzeman**
Closes #265 

- upgrade `aws-aplify` to v6
- update amplify setup to match new version
- implement custom `authFlowType` for static number for review process
- add comments and links to docs

In addition:
- replace `+` by placeholder "+421" in login screen input
- fix stuck flow on Notification permissions screen on ios simulator

TODOs:
- handle already signed in user properly
- handle wrong phone number format properly

### CODE REVIEW

File: app/(app)/settings/index.tsx [hunk 1]
```diff
@@ -35,16 +35,16 @@ const SettingsPage = () => {
...
     mutationFn: () => clientApi.usersControllerDeleteUser(),
     onSuccess: async (response) => {
       if (response.data) {
-        const deleteUserAWSResponse = await Auth.deleteUser()
-
-        if (deleteUserAWSResponse === 'SUCCESS') {
-          // signout to remove all data of user from local storage
-          await signOut()
-        }
+        await deleteUser()
```
Comment (PavelFilo) [inline app/(app)/settings/index.tsx:46]: Why did you remove the `SUCCESS` check?

File: hooks/useSignInOrSignUp.ts [hunk 2]
```diff
@@ -1,122 +1,151 @@
...
-        })
-        // TODO translation
-        console.log(`${GENERIC_ERROR_MESSAGE} - unexpected object thrown in confirmSignIn:`, error)
+        snackbar.show(t('unexpectedErrorObjectInConfirm'), { variant: 'danger' })
+        console.error(t('unexpectedErrorObjectInConfirm'), error)
       }
     }
   }
-
+  // eslint-disable-next-line unicorn/consistent-function-scoping
```
Comment (PavelFilo) [inline hooks/useSignInOrSignUp.ts:136]: what breakes this rule? the ternary?

File: package.json [hunk 3]
```diff
@@ -64,6 +63,7 @@
     "react-native-confirmation-code-field": "^7.3.2",
     "react-native-date-picker": "~4.3.3",
     "react-native-gesture-handler": "~2.12.0",
+    "react-native-get-random-values": "~1.9.0",
```
Comment (PavelFilo) [inline package.json:66]: I don't see it being used anywhere

Comment (radoslavzeman) [reply package.json:66]: It's from the docs 🤷‍♂️ https://docs.amplify.aws/react-native/build-a-backend/auth/set-up-auth/#install-amplify-libraries

**Review by PavelFilo — APPROVED** [review]
